test-generator.js

```diff
@@ -10,30 +13,58 @@ const formatOpts = {
   bracesSameLine: true
 };
 
-export const generateUnitTest = (msg) => {
-  if (!msg.data) {
+const leftHandDiff = (val) => {
```

Comment:
Because this function is a sequence of predicates and corresponding mappings over a single value, it's a really good case for [`cond()`](http://ramdajs.com/docs/#cond), which would look roughly like this:

```javascript
let leftHandDiff; leftHandDiff = cond([
  [isModifiedObject, prop('__old')],
  [isModifiedArray, pipe(
    filter(either(startsWith('~'), startsWith('-'))),
    map(slice(1, Infinity))
  )],
  [is(Array), map(leftHandDiff)],
  [is(Object), val => {
    const nonAddedKeys = pipe(keys, reject(endsWith('__added')))(val);
    const deepDive = pipe(map(propOf(val)), map(leftHandDiff));
    const newKeys = map(replace(/__deleted$/, ''), nonAddedKeys);
    return zipObj(newKeys, deepDive(nonAddedKeys));
  }]
]);
```

That's just a rough cut. You could make that last block point-free with a well-placed [`converge()`](http://ramdajs.com/docs/#converge).